### PR TITLE
MAILBOX-367 RabbitMQEvenBus draft impl

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import com.google.common.collect.ImmutableMap;
+import com.rabbitmq.client.AMQP;
+
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+public class RabbitMQEventBus implements EventBus {
+
+    private static final String MAILBOX_EVENT = "mailboxEvent";
+    static final String MAILBOX_WORK_QUEUE_NAME = MAILBOX_EVENT + "-exchange";
```
Comment (chibenwa) [inline mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java]: Why not private ?

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    private static final String MAILBOX_EVENT_EXCHANGE_NAME = MAILBOX_EVENT + "-workqueue";
```
Comment (chibenwa) [inline mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java]: Aren't you over anticipating next step by creating this queue?

(Maybe creating it in the test is enough...)

File: mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    static final boolean AUTO_ACK = true;
```
Comment (chibenwa) [inline mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java]: private?

Comment (chibenwa) [reply mailbox/event/event-rabbitmq/src/main/java/org/apache/james/mailbox/events/RabbitMQEventBus.java]: We may share these constants in rabbitmq-backend ?

File: mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import reactor.core.scheduler.Schedulers;
+
+class RabbitMQEventBusTest {
+
+    private static final MessageUid MESSAGE_UID = MessageUid.of(123456);
+    private static final Flags FLAGS = FlagsBuilder.builder()
+        .add(Flags.Flag.ANSWERED, Flags.Flag.DRAFT)
+        .add("User Custom Flag")
+        .build();
+    private static final MailboxListener.Added EVENT = new MailboxListener.Added(
```
Comment (chibenwa) [inline mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java]: I propose MailboxAdded which is simpler.

File: mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        assertThat(dequeueEvent().block())
+            .isEqualTo(EVENT);
+    }
+
+
+    private Mono<Event> dequeueEvent() {
+        RabbitMQChannelPool.RabbitFunction<GetResponse, IOException> dequeueRabbitEvent = channel ->
+            channel.basicGet(MAILBOX_WORK_QUEUE_NAME, !AUTO_ACK);
+
+        return reactor.core.publisher.Mono.fromSupplier(Throwing.supplier(() -> channelPool.execute(dequeueRabbitEvent)))
```
Comment (chibenwa) [inline mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java]: FQDN spotted...

**Comment by chibenwa** [comment]
```
[a3cf87e58b4dea13e841b301267c48c54384d574] [INFO] Running org.apache.james.mailbox.events.RabbitMQEventBusTest
[a3cf87e58b4dea13e841b301267c48c54384d574] SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
[a3cf87e58b4dea13e841b301267c48c54384d574] SLF4J: Defaulting to no-operation (NOP) logger implementation
[a3cf87e58b4dea13e841b301267c48c54384d574] SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
[a3cf87e58b4dea13e841b301267c48c54384d574]         ℹ︎ Checking the system...
[a3cf87e58b4dea13e841b301267c48c54384d574]         ✔ Docker version should be at least 1.6.0
[a3cf87e58b4dea13e841b301267c48c54384d574] [INFO] 
[a3cf87e58b4dea13e841b301267c48c54384d574] [INFO] Results:
[a3cf87e58b4dea13e841b301267c48c54384d574] [INFO] 
[a3cf87e58b4dea13e841b301267c48c54384d574] [INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
```

No results 

test this please

**Comment by chibenwa** [comment]
```
[a3cf87e58b4dea13e841b301267c48c54384d574] [INFO] Apache James :: Mailbox :: Event :: RabbitMQ implementation FAILURE [  02:00 h]
```

Fork timeout

**Comment by trantienduchn** [comment]
See full of content here #2070

**Pull request closed without merging**